src/partition/ParquetInterface.cc

```diff
@@ -0,0 +1,353 @@
+
+        // Increment the batch number
+        m_batchNumber++;
+
+        const arrow::FieldVector fields = initTable->schema()->fields();
+        for (auto fd : fields) {
+            fieldConfig[fd->name()] = fd;
+        }
+
+        arrow::FieldVector formatedTable_fields;
+        std::vector<std::shared_ptr<arrow::ChunkedArray>> formatedTable_columns;
+
+        // Loop over the column names as defined in the partition config file
+        for (std::string paramName : m_parameterNames) {
+            std::shared_ptr<arrow::ChunkedArray> chunkedArray = initTable->GetColumnByName(paramName);
+
+            // Column not found in the arrow table...
+            if (chunkedArray == NULLPTR) {
+                paramNotFound.push_back(paramName);
+            } else {
+                // Column type is boolean -> switch to 0/1 representation
+                if (fieldConfig[paramName]->type() == arrow::boolean()) {
+                    auto newChunkedArray = ChunkArrayReformatBoolean(chunkedArray, true);
+                    if (newChunkedArray == nullptr) {
+                        return arrow::Status::ExecutionError("Error while formating boolean chunk array");
+                    }
+                    formatedTable_columns.push_back(newChunkedArray);
+
+                    std::shared_ptr<arrow::Field> newField =
+                            std::make_shared<arrow::Field>(fieldConfig[paramName]->name(), arrow::int8());
+                    formatedTable_fields.push_back(newField);
+                }
+                // Simply keep the chunk as it is defined in teh arrow table
+                else {
+                    formatedTable_columns.push_back(chunkedArray);
+                    formatedTable_fields.push_back(fieldConfig[paramName]);
+                }
+            }
+        }  // end of loop over parameters
+
+        // If a column is not found (i.e. a parameter defined in partition.json does not exist in parquet
+        // file), throw an error and stop
+        if (paramNotFound.size() > 0) {
+            for (auto name : paramNotFound)
+                std::cout << "ERROR : param name " << name << " not found in table columns" << std::endl;
+            return arrow::Status::ExecutionError("Configuration file : missing parameter in table");
+        }
+
+        // Create the arrow::schema of the new table
+        std::shared_ptr<arrow::Schema> formatedSchema = std::make_shared<arrow::Schema>(
+                arrow::Schema(formatedTable_fields, initTable->schema()->endianness()));
+
+        // and finally create the arrow::Table that matches the partition config file
+        outputTable = arrow::Table::Make(formatedSchema, formatedTable_columns);
+        arrow::Status resTable = outputTable->ValidateFull();
+        if (!resTable.ok()) {
+            std::cout << "ERROR : formated table full validation not OK" << std::endl;
```

Comment:
`std::cout` is not allowed in Qserv code. Please, switch to using LSST Logging service.

Reply:
Done